parser.js

```diff
@@ -903,6 +903,62 @@ class DataflashParser {
         this.parseAtOffset(type)
         console.log('done')
     }
+
+    // Return array of objects giving stats about the composition of the log, sizes in bytes
+    stats() {
+
+        // Need this to access get_size_of in the helper function
+        const get_size_of_fun = this.get_size_of
+        function get_msg_size(Format) {
+            // Work out size of single message, all message have a 3 byte header
+            let size = 3
+            for (let i = 0; i < Format.length; i++) {
+                size += get_size_of_fun(Format.charAt(i))
+            }
+            return size
+        }
+
+        let total_size_check = 0
+        const typeSet = new Set(this.msgType)
+        let ret = {}
+        for (const msg of this.FMT) {
+            if (msg && typeSet.has(msg.Type)) {
+                // Count occurrences
+                let count = 0
+                for (const type of this.msgType) {
+                    if (type == msg.Type) {
+                        count++
+                    }
+                }
+
+
+                const msg_size = get_msg_size(msg.Format)
+                const size = msg_size * count
+                ret[msg.Name] = { count, msg_size, size }
+
+                total_size_check += size
+            }
+        }
+
+        // Total should match log size
```

Comment:
This is only really a sanity check, we could remove it. Or add the difference as a item "corrupted".

Reply:
I thnk we should do a warn instead of log. but I think it makes sense to keep it.

Reply:
Changed to warn and removed the early return. So you get the warning but you still get some stats.